ISPN-5380 Internal Cache Registry

tristantarrant opened the pull request:
https://issues.jboss.org/browse/ISPN-5380
https://issues.jboss.org/browse/ISPN-5381
https://issues.jboss.org/browse/ISPN-5382

gustavonalle commented on core/src/main/java/org/infinispan/registry/impl/InternalCacheRegistryImpl.java:
```diff
@@ -0,0 +1,61 @@
+public class InternalCacheRegistryImpl implements InternalCacheRegistry {
+   private static final Log log = LogFactory.getLog(MethodHandles.lookup().lookupClass());
+   private EmbeddedCacheManager cacheManager;
+   private final Set<String> internalCaches = new HashSet<>();
```
Does it need a concurrent collection here?

tristantarrant replied:
Yes, better stay on the safe side. I guess I can just use a ConcurrentHashMap<String, Boolean> where the value is meaningless

gustavonalle replied:
There's a ```ConcurrentHashSet``` internally you could use as well

tristantarrant commented:
@gustavonalle changed it

tristantarrant commented:
Dropped ISPN-5383 commit and issued a new PR just for that

gustavonalle commented:
pulling

anistor commented:
@tristantarrant @gustavonalle Shouldn't we use the internal cache registry for the QueryCache too?

gustavonalle commented:
I'm picking some failures on ```client/hotrod-client```:

```org.infinispan.server.hotrod.CacheNotFoundException: Cache with name '___script_cache' not found amongst the configured caches```

```org.infinispan.server.hotrod.CacheNotFoundException: Cache with name '___protobuf_metadata' not found amongst the configured caches```

do you see them as well?

tristantarrant commented:
Let me check

On Fri, Apr 17, 2015 at 1:05 PM, Gustavo Fernandes <notifications@github.com
> wrote:

> I'm picking some failures on client/hotrod-client:
>
> org.infinispan.server.hotrod.CacheNotFoundException: Cache with name
> '___script_cache' not found amongst the configured caches
>
> org.infinispan.server.hotrod.CacheNotFoundException: Cache with name
> '___protobuf_metadata' not found amongst the configured caches
>
> do you see them as well?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/infinispan/infinispan/pull/3386#issuecomment-93965614>
> .
>

anistor commented:
I get the same errors, with or without rebasing this to latest.

tristantarrant commented:
I added the concept of "USER" caches, which are internal caches which are user-visible. And I've also moved the logic of filtering out internal caches to the registry itself.

gustavonalle commented:
@anistor the QueryCache could use this too. Do you see any issue of doing it in a separate jira?

anistor commented:
@gustavonalle I'll add a jira for using the internal cache registry for QueryCache

anistor commented:
@gustavonalle Just added an issue to further track it: https://issues.jboss.org/browse/ISPN-5390

gustavonalle commented:
merged. tks @tristantarrant
